lua/obsidian/command.lua

```diff
@@ -216,6 +216,123 @@ command.search = function(client, data)
+command.insert_template = function(client, data)
+  if not client.opts.templates.subdir then
+    echo.err "No templates folder defined in setup()"
+    return
+  end
+
+  local templates_dir = Path:new(client.dir) / client.opts.templates.subdir
+  if not templates_dir:is_dir() then
+    echo.err(string.format("%s is not a valid directory for templates", templates_dir))
+    return
+  end
+
+  -- We need to get these upfront otherwise
+  -- Telescope hijacks the current window
+  local buf = vim.api.nvim_win_get_buf(0)
+  local win = vim.api.nvim_get_current_win()
+  local row, _ = unpack(vim.api.nvim_win_get_cursor(win))
+
+  local apply_template = function(name)
+    local template_path = Path:new(templates_dir / name)
+    local date_format = client.opts.templates.date_format or "%Y-%m-%d"
+    local time_format = client.opts.templates.time_format or "%H:%M"
+    local date = tostring(os.date(date_format))
+    local time = tostring(os.date(time_format))
+    local current_fp = vim.api.nvim_buf_get_name(buf)
+    local _, _, title = string.find(vim.fs.normalize(current_fp), ".*/(.-)%.md")
+    title = title or vim.fs.basename(current_fp)
+
+    local insert_lines = {}
+    local template_file = io.open(tostring(template_path), "r")
+    if template_file then
+      local lines = template_file:lines()
+      for line in lines do
+        line = string.gsub(line, "{{date}}", date)
+        line = string.gsub(line, "{{time}}", time)
+        line = string.gsub(line, "{{title}}", title)
+        table.insert(insert_lines, line)
+      end
+      template_file:close()
+    end
+
+    vim.api.nvim_buf_set_lines(buf, row - 1, row - 1, false, insert_lines)
+  end
+
+  -- try with telescope.nvim
+  local has_telescope, _ = pcall(require, "telescope.builtin")
+  if has_telescope then
+    local choose_template = function()
+      local opts = {
+        cwd = tostring(templates_dir),
+        attach_mappings = function(_, map)
+          map({ "i", "n" }, "<CR>", function(prompt_bufnr)
+            local template = require("telescope.actions.state").get_selected_entry()
+            apply_template(template[1])
+            require("telescope.actions").close(prompt_bufnr)
+          end)
+          return true
+        end,
+      }
+      require("telescope.builtin").find_files(opts)
+    end
+    choose_template()
+    return
+  end
+
+  -- try with fzf-lua
+  local has_fzf_lua, fzf_lua = pcall(require, "fzf-lua")
+  if has_fzf_lua then
+    local cmd = vim.tbl_flatten { util.FIND_CMD, { ".", "-name", "'*.md'" } }
+    cmd = util.table_params_to_str(cmd)
+    fzf_lua.files {
+      cmd = cmd,
+      cwd = tostring(templates_dir),
+      file_icons = false,
+      actions = { ["default"] =
+      function(entry)
+        -- for some reason fzf-lua passes the filename with 6 characters
+        -- at the start that appear on screen as 2 whitespace characters
+        -- so we need to start on the 7th character
+        local template = entry[1]:sub(7)
```

Comment:
If those are all whitespace characters it would probably be more robust to trim off the whitespace using gsub or match. Here are some example implementations: http://lua-users.org/wiki/StringTrim

Reply:
Tried these, doesn't work. It's a very strange behaviour from fzf-lua I think

Reply:
Weird.. I guess those characters aren't typical whitespace?

Reply:
Yup, I'm also struggling to reproduce it in another script to try and isolate the problem.
```lua
local fzf_lua = require("fzf-lua")
fzf_lua.files {
  cmd = "find -name '*.lua'",
  file_icons = false,
  git_icons = false,
  actions = { ["default"] =
  function(entry)
    print(#entry[1])
  end
  }
}
```
Sourcing this and picking a file is printing the correct number of characters, no weird extra spaces.